springbackend/src/main/java/com/simplecoding/social/controller/PrivateEndpoint.java

```diff
@@ -68,4 +75,17 @@
         return new ResponseEntity<List<User>>(myFriends, HttpStatus.OK);
     }
 
+    @PostMapping("addPost")
+    public ResponseEntity<?> addPost(@RequestBody Post post) throws NullPointerException {
+        UserDto user = securityService.getUser();
+        Post savedPost = postService.savePost(user,post.getContent());
+        return ResponseEntity.created(URI.create("/private/post")).body(savedPost);
+    }
+
+    @GetMapping("mypost")
```

Comment:
make it /user_id/posts and move to public, no authentication needed for this api

Reply:
this api is for loggedin users only so they can see  posts which are posted by them.
ill create another public api for listing  all posts.